tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Generators/ReadMeGeneratorTool.cs

```diff
@@ -0,0 +1,285 @@
+                    * A package path that we can use to generate documentation links
+                    Here are some more rules to follow:
+                    - Do not touch the following sections, or its subsections: Contributing.
+                    - Do not generate sample code.
+                    - Rules for proper readmes can be found here: https://github.com/Azure/azure-sdk/blob/main/docs/policies/README-TEMPLATE.md
+                    """
+                ),
+                new SystemChatMessage($"The readme template is this: {readmeText} which we fill in with the user parameters, which follow:"),
+                new UserChatMessage($"Service URL: {serviceDocumentation}"),
+                new UserChatMessage($"Package path: {packagePath}")
+            };
+
+            var response = await chatClient.CompleteChatAsync(messages);
+            var generatedReadmeText = Customize(response.Value.Content[0].Text);
+
+            await File.WriteAllTextAsync(outputPath, generatedReadmeText, ct);
+            output.Output($"Readme written to {outputPath}");
+
+            return await IsValid(repoPath: repoPath, readmePath: outputPath, ct: ct);
+        }
+
+        private async Task<bool> IsValid(string repoPath, string readmePath, CancellationToken ct)
+        {
+            var valid = true;
+
+            try
+            {
+                var result = await VerifyLinks(repoPath, readmePath, ct);
+
+                if (result != null)
+                {
+                    output.OutputError($"[FAIL] Verify-Links.ps1 failed: {result}");
+                    valid = false;
+                }
+                else
+                {
+                    output.Output($"[PASS] Verify-Links.ps1 passed");
+                }
+            }
+            catch (Exception ex)
+            {
+                output.OutputError($"[ERROR] Verify-Links.ps1 threw an exception: {ex.Message}");
+                valid = false;
+            }
+
+            return valid;
+        }
+
+        /// <summary>
+        /// Run the Verify-Links.ps1 script, which checks that all links in the readme are valid.
+        /// </summary>
+        /// <param name="repoPath">Path to an azure-sdk-for-<language> repository</language></param>
+        /// <param name="readmePath">Path to README.md to verify</param>
+        /// <param name="ct">Cancellation token for method</param>
+        /// <returns>A string, if the verification did not pass, or null if it did.</returns>
+        /// <exception cref="Exception"></exception>
+        private async Task<string> VerifyLinks(string repoPath, string readmePath, CancellationToken ct)
+        {
+            var verifyLinksPs1 = Path.Join(repoPath, "eng", "common", "scripts", "Verify-Links.ps1");
+            var errors = new List<string>();
+
+            output.Output($"Running {verifyLinksPs1} {readmePath}");
+
+            var process = Process.Start(new ProcessStartInfo()
```

Comment:
We can refactor this later into the process service that @chrisradek has in PR right now.